Libraries/LibWeb/IndexedDB/Internal/Algorithms.cpp

```diff
@@ -1213,4 +1213,24 @@ GC::Ref<IDBRequest> asynchronously_execute_a_request(JS::Realm& realm, IDBReques
     return request;
 }
 
+// https://w3c.github.io/IndexedDB/#generate-a-key
+WebIDL::ExceptionOr<u64> generate_a_key(JS::Realm& realm, GC::Ref<ObjectStore> store)
+{
+    // 1. Let generator be store’s key generator.
+    auto generator = store->key_generator().value();
+
+    // 2. Let key be generator’s current number.
+    auto key = generator.current_number();
+
+    // 3. If key is greater than 2^53 (9007199254740992), then return failure.
+    if (key > 9007199254740992)
+        return WebIDL::ConstraintError::create(realm, "Key is greater than 2^53"_string);
```

Comment:
here you converted the spec text `return failure` into a WebIDL Exception type. Why not also do that in the places you return `ErrorOr<T>`  earlier in the patch series? I feel like that would help resolve Kenneth's concerns as well. Though it's also true that this "failure" concept likely needs some real editor review globally.

Reply:
Returning a WebIDL directly here was done since the only caller of this specifies this exception. 
Since its not what the spec wants at this point, its been changed to use a sentinel error value instead.